### ISSUE TITLE
Refactor to use uv

### ISSUE THREAD

**Issue by chrisrodrigue** [winpython/winpython#1420]
uv can create virtual environments that are binary compatible to those created by virtualenv. They can be made with the `--relocatable` flag, which provides true portability (you can move your distribution anywhere, including other Windows PCs, without relying on path or shell hacks). This could allow a for a significant refactor of WinPython! 

It also replaces pip, pip-tools, pipx, poetry, pyenv,
and will even be getting its own build backend! See https://github.com/astral-sh/uv/issues/3957 for more details on that. As a static binary written in rust, it has 0 installed or transient dependencies and will significantly reduce the size of the WinPython distribution while providing speed and features to users. It is shaping up to be the defacto Python toolchain and dependency manager (like cargo, but for Python).

I would highly recommend considering it for inclusion.

**Comment by stonebig** [comment]
uv is still:
- doing 3 release per week as oja Nauray 2nd 2025, which is "odd" for a package manager
- playing way over a simple package manager.

It's not sure it fits with WinPython and WinPython type of users, Psychologically  speaking

**Comment by chrisrodrigue** [comment]
Yeah, they are on a very fast release schedule right now, but they are very good about not introducing breaking changes (unlike poetry). 

uv can also manage python installations and function as a build backend. If you are familiar with Rust, they are attempting to become the "cargo" for Python (one tool to rule them all). 

FWIW, they are good about listening to users and adding requested features. Same team as the ruff formatter/linter.
